source/transactions/tests/commit.yml

```diff
@@ -455,3 +455,88 @@ tests:
     outcome:
       collection:
         data: []
+
+  - description: reset session state
+
+    operations:
+      - name: startTransaction
+        arguments:
+          session: session0
+      - name: insertOne
+        arguments:
+          document:
+            _id: 1
+          session: session0
+        result:
+          insertedId: 1
+      - name: commitTransaction
+        arguments:
+          session: session0
+      # Running any operation after an ended transaction resets the session
+      # state to "no transaction".
+      - name: insertOne
+        arguments:
+          document:
+            _id: 2
+          session: session0
+        result:
+          insertedId: 2
+      # Calling commit again should error instead of re-running the commit.
+      - name: commitTransaction
+        arguments:
+          session: session0
+        result:
+          errorContains: no transaction started
+      # Calling abort should also error with "no transaction started".
+      - name: abortTransaction
+        arguments:
+          session: session0
+        result:
+          errorContains: no transaction started
```

Comment:
Let's split this test in two. One test does start, insert, commit, insert, commit, and the other does start, insert, commit, insert, **abort**.

I'm concerned in this test that calling commit, then abort, muddies the water. We know that insert should reset the state, and then calling "commit" and getting an error proves that the state is reset. But then we call abort, and the state of the transaction at that point could be the result of the insert call **or** the commit call. So let's split up the test to be more precise.

Reply:
Update with two tests almost as you've described. The first does commit, insert, commit and the second does abort, insert, abort.